app/graphql/mutations/destroy_record.rb

```diff
@@ -2,35 +2,46 @@
 
 module Mutations
   class DestroyRecord < BaseMutation
-    argument :id, Integer, required: true
+    argument :id, Integer, required: false
     argument :record_type, String, required: true
+    argument :external_id, Integer, required: false
 
     type Types::DestroyType
 
     RECORD_WHITELIST = ["EventRecord", "NewsItem", "PointOfInterest", "Tour"].freeze
 
-    def resolve(id:, record_type:)
+    def resolve(id: nil, record_type:, external_id: nil)
       return error_status unless RECORD_WHITELIST.include?(record_type)
 
-      record = record_type.constantize
-                 .filtered_for_current_user(context[:current_user])
-                 .where(id: id)
-                 .first
+      record = if external_id.present?
+                 find_record(record_type, external_id: external_id)
+               else
+                 find_record(record_type, id: id)
+               end
+      result = destroy_record(record)
+      id ||= record.try(:id)
 
-      if record.present?
-        record.destroy
-        status = "Record destroyed"
-        status_code = 200
-      else
-        status = "Record not found"
-        status_code = 404
-      end
-
-      OpenStruct.new(id: id, status: status, status_code: status_code)
+      OpenStruct.new(id: id, status: result[:status], status_code: result[:status_code])
     end
 
     def error_status
       OpenStruct.new(id: nil, status: "Access not permitted", status_code: 403)
     end
+
```

Comment:
die methoden hier unten könnten noch in ein `private`, oder?

Reply:
Ja.